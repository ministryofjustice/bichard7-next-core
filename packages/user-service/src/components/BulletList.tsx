import React from "react"

interface BulletListProps {
  heading?: string
  items: string[]
  listClassName?: string
  headingClassName?: string
}

const BulletList: React.FC<BulletListProps> = ({
  heading,
  items,
  listClassName = "govuk-list govuk-list--bullet",
  headingClassName = "govuk-body"
}) => {
  if (!items || items.length === 0) {
    return null
  }

  return (
    <>
      {heading && <p className={headingClassName}>{heading}</p>}

      <ul className={listClassName}>
        {items.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
    </>
  )
}

export default BulletList
