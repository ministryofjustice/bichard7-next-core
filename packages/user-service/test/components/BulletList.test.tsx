import { render } from "@testing-library/react"
import BulletList from "components/BulletList"

describe("BulletList", () => {
  it("should render the heading and list items and match the snapshot", () => {
    const items = ["apples", "oranges", "pears"]

    const { container } = render(<BulletList heading="You can buy:" items={items} />)

    expect(container).toMatchSnapshot()
  })

  it("should render only the list items when no heading is provided", () => {
    const items = ["A", "B", "C"]

    const { container } = render(<BulletList items={items} />)

    expect(container).toMatchSnapshot()
  })

  it("should render null/nothing when the items array is empty", () => {
    const { container } = render(<BulletList heading="Empty List" items={[]} />)

    expect(container).toMatchSnapshot()
  })
})
