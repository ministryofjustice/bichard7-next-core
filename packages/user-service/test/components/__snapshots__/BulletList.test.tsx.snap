// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BulletList should render null/nothing when the items array is empty 1`] = `<div />`;

exports[`BulletList should render only the list items when no heading is provided 1`] = `
<div>
  <ul
    class="govuk-list govuk-list--bullet"
  >
    <li>
      A
    </li>
    <li>
      B
    </li>
    <li>
      C
    </li>
  </ul>
</div>
`;

exports[`BulletList should render the heading and list items and match the snapshot 1`] = `
<div>
  <p
    class="govuk-body"
  >
    You can buy:
  </p>
  <ul
    class="govuk-list govuk-list--bullet"
  >
    <li>
      apples
    </li>
    <li>
      oranges
    </li>
    <li>
      pears
    </li>
  </ul>
</div>
`;
